### PR TITLE
Assistant V2: Patch Assistant route

### PR DESCRIPTION
**Pull request by PopDaph**
Adding Assistant API route `POST /api/w/[wId]/assistants/[aId] update assistant`. -> I used a patch over a POST, is that okay?
Part of https://github.com/dust-tt/dust/issues/1347

Not tested!

### CODE REVIEW

File: front/pages/api/w/[wId]/assistants/index.ts [hunk 1]
```diff
@@ -171,7 +171,8 @@ async function handler(
         status_code: 405,
         api_error: {
           type: "method_not_supported_error",
-          message: "The method passed is not supported, GET is expected.",
+          message:
+            "The method passed is not supported, GET OR POST is expected.",
```
Comment (PopDaph) [inline front/pages/api/w/[wId]/assistants/index.ts:175]: I forgot to update that string on the last PR

File: front/pages/api/w/[wId]/assistants/[aId]/index.ts [hunk 2]
```diff
@@ -11,6 +18,74 @@ export type GetAssistantResponseBody = {
   assistant: AgentConfigurationType;
 };
 
+const PatchAssistantResponseBodySchema = t.type({
```
Comment (spolu) [inline front/pages/api/w/[wId]/assistants/[aId]/index.ts:21]: Should we just reuse the time from the creation endpoint?

**Review by spolu — DISMISSED** [review]

File: front/pages/api/w/[wId]/assistants/[aId]/index.ts (hunk 2 above)
Comment (PopDaph) [reply front/pages/api/w/[wId]/assistants/[aId]/index.ts]: I added a commit!

File: front/pages/api/w/[wId]/assistants/[aId]/index.ts [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
+  updateAgentActionConfiguration,
+  updateAgentConfiguration,
+  updateAgentGenerationConfiguration,
+} from "@app/lib/api/assistant/configuration";
 import { Authenticator, getSession } from "@app/lib/auth";
 import { ReturnedAPIErrorType } from "@app/lib/error";
 import { apiError, withLogging } from "@app/logger/withlogging";
 import { AgentConfigurationType } from "@app/types/assistant/agent";
 
+import { PostOrPatchAssistantResponseBodySchema } from "..";
```
Comment (spolu) [inline front/pages/api/w/[wId]/assistants/[aId]/index.ts:17]: relative import :grimacing: :)

**Pull request merged**